sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +112,31 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, 
+      complainFun: (Boolean, String) => Unit): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val rapidsJarURLs = classloader.getResources(propName).asScala.toSet
+    lazy val rapidsJars = rapidsJarURLs.map(_.toString.split("!").head).mkString(",")
+    lazy val rapidsJarsVers = rapidsJarURLs.map { 
+      url => scala.io.Source.fromInputStream(url.openStream()).mkString("") 
+    }.mkString(",")
+    complainFun(rapidsJarURLs.size <= 1,
+        s"Multiple $jarName jars found in the classpath: $rapidsJars, please make sure there " +
```

Comment:
cudf jar is no longer required, do we need to document this here ?

Reply:
Seems the cuDF jar inside the plugin jar will be detected by this function, so I think it will be fine if it detect only one cuDF jar.

An experiment:
1. Change the `complainFun` to just `logWarning`.
2. build plugin normally.
3. `export SPARK_RAPIDS_PLUGIN_JAR=spark-rapids/dist/target/rapids-4-spark*.jar`
4. `spark-shell --master local[*] --jars ${SPARK_RAPIDS_PLUGIN_JAR} --conf spark.plugins=com.nvidia.spark.SQLPlugin --conf spark.rapids.sql.allowMultipleJars=true`

Result:
```
23/11/13 17:34:45 WARN RapidsPluginUtils: Multiple rapids4spark jars found in the classpath: jar:file:/home/haoyangl/spark-rapids/dist/target/rapids-4-spark_2.12-23.12.0-SNAPSHOT-cuda11.jar, please make sure there is only one rapids4spark jar in the classpath. If it is impossible to fix the classpath you can suppress the error by setting spark.rapids.sql.allowMultipleJars to true. Version info:
version=23.12.0-SNAPSHOT
cudf_version=23.12.0-SNAPSHOT
user=haoyangl
revision=14c536d6ba94d64c5468d711624449c452e54224
branch=detect_multiple_jars
date=2023-11-13T09:22:21Z
url=https://github.com/thirtiseven/spark-rapids.git

23/11/13 17:34:45 WARN RapidsPluginUtils: Multiple spark-rapids-jni jars found in the classpath: jar:file:/home/haoyangl/spark-rapids/dist/target/rapids-4-spark_2.12-23.12.0-SNAPSHOT-cuda11.jar, please make sure there is only one spark-rapids-jni jar in the classpath. If it is impossible to fix the classpath you can suppress the error by setting spark.rapids.sql.allowMultipleJars to true. Version info:
version=23.12.0-SNAPSHOT
user=
revision=c63544ec55b47a1660f0ff7d26ad55630133ad03
branch=HEAD
date=2023-11-12T07:13:35Z
url=https://github.com/NVIDIA/spark-rapids-jni.git

23/11/13 17:34:45 WARN RapidsPluginUtils: Multiple cudf jars found in the classpath: jar:file:/home/haoyangl/spark-rapids/dist/target/rapids-4-spark_2.12-23.12.0-SNAPSHOT-cuda11.jar, please make sure there is only one cudf jar in the classpath. If it is impossible to fix the classpath you can suppress the error by setting spark.rapids.sql.allowMultipleJars to true. Version info:
version=23.12.0-SNAPSHOT
user=
revision=87d2a36f04f431a8c5236d2aee723ec79b9dc5f9
branch=HEAD
date=2023-11-12T07:13:35Z
url=https://github.com/rapidsai/cudf.git
```